helperAPI.py

```diff
@@ -338,47 +338,70 @@ def get_result(self):
         return self.queue.get()
 
 
+def isUpToDate(remote, branch):
+    # Assume succeeded in updater()
+    import git
+
+    # Check if local branch is up to date with ls-remote
+    up_to_date = False
+    try:
+        g = git.cmd.Git()
+        ls_remote = g.ls_remote(remote, branch)
+        remote_hash = ls_remote.split("\t")[0]
+        local_commit = git.Repo(".").head.commit.hexsha
+        if str(local_commit) == str(remote_hash):
+            up_to_date = True
+            print(f"You are up to date with {remote}/{branch}")
+    except Exception as e:
+        print(f"Error running ls-remote: {e}")
+    if not up_to_date:
+        if remote_hash == "":
+            remote_hash = "NOT FOUND"
+        print(
+            f"WARNING: YOU ARE OUT OF DATE. Please run 'git pull {remote} {branch}' to update. Local hash: {local_commit}, Remote hash: {remote_hash}"
+        )
+    return up_to_date
+
+
 def updater():
     # Check if git is installed
     try:
         import git
-        from git import Repo
     except ImportError:
         print(
             "UPDATE ERROR: Git is not installed. Please install Git and then run pip install -r requirements.txt"
         )
         print()
         return
-    print("Starting auto update...")
+    print("Starting Git auto update...")
     try:
-        repo = Repo(".")
+        repo = git.Repo(".")
     except git.exc.InvalidGitRepositoryError:
         # If downloaded as zip, repo won't exist, so create it
-        repo = Repo.init(".")
+        repo = git.Repo.init(".")
         repo.create_remote("origin", "https://github.com/NelsonDane/auto-rsa")
         repo.remotes.origin.fetch()
         # Always create main branch
         repo.create_head("main", repo.remotes.origin.refs.main)
         repo.heads.main.set_tracking_branch(repo.remotes.origin.refs.main)
-        # If downloaded from other branch, zip has branch name
-        current_dir = Path.cwd()
-        if current_dir.name != "auto-rsa-main":
-            branch = str.replace(current_dir.name, "auto-rsa-", "")
+        repo.heads.main.checkout(True)
+        # When downloaded as zip, it contains the branch name
+        branch = str(Path.cwd().name).split("-")[-1]
+        if branch != "main":
             try:
                 repo.create_head(branch, repo.remotes.origin.refs[branch])
                 repo.heads[branch].set_tracking_branch(repo.remotes.origin.refs[branch])
                 repo.heads[branch].checkout(True)
             except:
```

Comment:
I don't know Python, is `except:` or `except Exception:` more acceptable?

Reply:
`except Exception` is the better practice, thanks for the catch :wink: